src/hyperion/device_setup_plans/read_hardware_for_setup.py

```diff
@@ -18,13 +19,15 @@ def read_hardware_for_ispyb_pre_collection(
     undulator: Undulator,
     synchrotron: Synchrotron,
     s4_slit_gaps: S4SlitGaps,
+    aperture_scatterguard:ApertureScatterguard
 ):
     LOGGER.info("Reading status of beamline parameters for ispyb deposition.")
     yield from bps.create(
         name=ISPYB_HARDWARE_READ_PLAN
     )  # gives name to event *descriptor* document
     yield from bps.read(undulator.current_gap)
     yield from bps.read(synchrotron.machine_status.synchrotron_mode)
+    yield from bps.read(aperture_scatterguard.aperture_name)
```

Comment:
Must: As I said in the `dodal` PR I'm not sure if reading a non-signal like this works. Could you add/update one of the tests in Hyperion to confirm that after the name is changed it can be read and will be put into the event document correctly?

Reply:
ok, didn't realize that properties are signals or non-signals

Reply:
the `_update_name` private method is run on every invokation of `set` that's not an error. Maybe on error it should do a check and set the position as invalid
![image](https://github.com/DiamondLightSource/hyperion/assets/56644812/23b52e59-f025-4ff9-b2c0-7d3b14e13b72)

Reply:
resolved through addition of signal interface